File: src/test/java/com/evan/dynamicprogramming/dp/robTest.java
package com.evan.dynamicprogramming.dp;

import com.evan.dynamicprogramming.dp.rob.Solution;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

public class robTest {

    @Test
    public void test(){
        Solution solution = new Solution();
        int[] nums = {1,2,3,1};
        Assertions.assertEquals(4,solution.rob(nums));
    }

    @Test
    public void test2(){
        Solution solution = new Solution();
        int[] nums = {2,7,9,3,1};
        Assertions.assertEquals(12,solution.rob(nums));
    }
}

File: src/test/kotlin/dp/rob/robTest.kt
package dp.rob

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

class robTest {

    @Test
    fun test(){
        val solution: SolutionKt = SolutionKt()
        val nums = intArrayOf(2,3,2)
        Assertions.assertEquals(3, solution.rob(nums))
    }

    @Test
    fun test2(){
        val solution: SolutionKt = SolutionKt()
        val nums = intArrayOf(2,3)
        Assertions.assertEquals(3, solution.rob(nums))
    }

    @Test
    fun test3(){
        val solution: SolutionKt = SolutionKt()
        val nums = intArrayOf(3)
        Assertions.assertEquals(3, solution.rob(nums))
    }

    @Test
    fun test4(){
        val solution: SolutionKt = SolutionKt()
        val nums = intArrayOf(1,2,3)
        Assertions.assertEquals(3, solution.rob(nums))
    }

    @Test
    fun test5(){
        val solution: SolutionKt = SolutionKt()
        val nums = intArrayOf(1,2,3,1)
        Assertions.assertEquals(4, solution.rob(nums))
    }
}